add death and other tracker

Koda-Dog opened the pull request:
This will add the ability to track also death by a player and other ways to die.

I’ve insert 3 Sliders that you can choose what you prefer to track.

API will still only send kill events.

CSV-Structure is changed but the script makes a backup of the old file and converts the data to the new structure.
I have evaluated it already some hours in AC. Seems fine now. But I was not able to evaluate it in 4.0 cause 4.0 isn’t running.

Koda-Dog committed:
Added dath and other tracking

It is now possible to record death and other fatal events by setting a slider.

The CSV structure has been changed but this updates the script automatically.

Api still only sends in case of a kill event.

Koda-Dog committed:
Some bugfixes

‘unknown’ deathmetode exception.
OtherLogSlider moved.
Wrong Ship output while death event.

Koda-Dog committed:
#Improvements for better maintenance in the script

## Summary
- Added structured functions for better readability and maintenance.
- Improved error handling using `Write-Warning` and `Write-Error`.
- Debug handling for Write-Output in script
- Increased modularity through separate functions.
- Implemented caching to improve performance.
- Fixed multiple bugs affecting data processing.
- Add functionality to prepend new log entries to the DebugPanel

## Changes
- **Structure**: Defined functions like `Send-ApiData`, `Write-CSVData`, and `New-KillEvent`.
- **Error Handling**: Implemented comprehensive error handling.
- **Modularity**: Functions are now modular and reusable.
- **Caching**: Added caching to reduce redundant computations and enhance performance.
- **Bug Fixes**: Resolved issues with data processing. Ensuring more reliable execution.

Koda-Dog committed:
Update KillTrackR_MainScript.ps1

forgot to finish implement "function Send-ApiData"

Koda-Dog commented:
Unfortunately, I do not have an API server to test the ‘Send-ApiData function’ in KillTrackR_MainScript.ps1. Someone else should do that. And by the way, thanks for all the work for this software, really great!

Koda-Dog committed:
Update HomePage.xaml.cs

Implement fallback for profile image URL when primary URL is empty

Koda-Dog committed:
Update KillTrackR_MainScript.ps1

-Enhance logic to correctly handle time-based expiration for last kill tracking for identifiying a Passenger

-PvE event detection and handling reimplemented

Koda-Dog committed:
Update KillTrackR_MainScript.ps1

Koda-Dog committed:
Optimizations and Bug Fixes

- Optimized structure of type detection
- Implemented cache for HTML requests
- Improved handling of 'Other' events
- Fixed a bug with renaming video files

Koda-Dog committed:
BugFix

- solve a Bug with other way to die

Heavybob commented:
> Unfortunately, I do not have an API server to test the ‘Send-ApiData function’ in KillTrackR_MainScript.ps1. Someone else should do that. And by the way, thanks for all the work for this software, really great!

The API server isn't too complicated. Just use express for testing. https://expressjs.com/

If you need an api key, I've sent a dm on discord.

BubbaGumpShrump commented:
Hey man, I appreciate everything you've done. I'm sorry I've been out of contact for a bit. Code looks great reading through and I'd like to merge it and then fix a few things that broke in the latest patch a day or so ago.

Do you have any plans for the "OtherLog" toggle? Would you be okay with us making that a "LocationData" toggle?

Koda-Dog commented:
So I don't have any plans for the OtherLog changer in the future. However, I think the current function is quite important, as surely not everyone wants videos of their crashes or wants to know how often they have a “suicide” event. I think it would be even better to do without the KillLog toggle as this is the function that is currently always used.

Koda-Dog committed:
Update to 2.07 change to GameMode

-Code update to 2.07
-Instead of saving GameMode ‘AC’, the original AC-Mode key is now used

Koda-Dog committed:
refactor: Combine event handling for events

Refactored the code to consolidate the handling of events into a single reusable method, `HandleKillEvent`.

- Extracted shared logic into `HandleKillEvent`.
- Added dynamic adjustments for specific event types.

Koda-Dog committed:
Improve error handling

- Enhanced error catching from the PowerShell script to ensure robust execution.

Heavybob commented:
Appreciate the changes you contributed koda. https://github.com/BubbaGumpShrump/AutoTrackR2/pull/9 is going to be eventually pulled which will depreciate powershell entirely meaning these changes would need to be adapted.

Koda-Dog commented:
I'm glad to have contributed something. Looking forward to the new version. Close Pull.
